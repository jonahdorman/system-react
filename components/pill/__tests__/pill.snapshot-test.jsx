/* eslint-env jest */
import React from 'react';
import { renderMarkup, testDOMandHTML } from '../../../tests/snapshot-helpers';

import BaseExample from '../__examples__/base';
import IconExample from '../__examples__/icon';

testDOMandHTML({
	name: 'Linked, Unlinked, Truncated',
	test,
	Component: BaseExample,
});

testDOMandHTML({
	name: 'Icon, Avatar, Error',
	test,
	Component: IconExample,
});
