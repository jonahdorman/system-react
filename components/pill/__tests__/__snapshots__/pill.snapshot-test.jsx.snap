// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon, Avatar, Error DOM Snapshot 1`] = `
<div
  className="slds-grid slds-grid_pull-padded-medium"
>
  <div
    className="slds-p-horizontal_medium"
  >
    <span
      className="slds-pill slds-pill_link"
      onClick={null}
      onKeyDown={[Function]}
      role="button"
    >
      <span
        className="slds-pill__icon_container"
      >
        <span
          className="slds-icon_container slds-icon-standard-account"
          title="Account"
        >
          <svg
            aria-hidden="true"
            className="slds-icon"
          >
            <use
              xlinkHref="/assets/icons/standard-sprite/svg/symbols.svg#account"
            />
          </svg>
          
        </span>
      </span>
      <a
        className="slds-pill__action"
        href="javascript:void(0);"
        onClick={[Function]}
        title="Full pill label verbiage mirrored here"
      >
        <span
          className="slds-pill__label"
        >
          Pill Label
        </span>
      </a>
      <span
        className="slds-icon_container slds-pill__remove"
        onClick={[Function]}
        role="button"
        title="Remove"
      >
        <svg
          className="slds-icon slds-icon_x-small slds-icon-text-default"
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Remove
        </span>
      </span>
    </span>
  </div>
  <div
    className="slds-p-horizontal_medium"
  >
    <span
      className="slds-pill slds-pill_link"
      onClick={null}
      onKeyDown={[Function]}
      role="button"
    >
      <span
        className="slds-pill__icon_container"
      >
        <div>
          <span
            className="slds-avatar slds-avatar_circle slds-avatar_medium"
          >
            <img
              alt=""
              onError={[Function]}
              src="https://lightningdesignsystem.com/assets/images/avatar2.jpg"
              title="User avatar"
            />
          </span>
        </div>
      </span>
      <a
        className="slds-pill__action"
        href="javascript:void(0);"
        onClick={[Function]}
        title="Full pill label verbiage mirrored here"
      >
        <span
          className="slds-pill__label"
        >
          Pill Label
        </span>
      </a>
      <span
        className="slds-icon_container slds-pill__remove"
        onClick={[Function]}
        role="button"
        title="Remove"
      >
        <svg
          className="slds-icon slds-icon_x-small slds-icon-text-default"
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Remove
        </span>
      </span>
    </span>
  </div>
  <div
    className="slds-p-horizontal_medium"
  >
    <span
      className="slds-pill slds-pill_link slds-has-error"
      onClick={null}
      onKeyDown={[Function]}
      role="button"
    >
      <span
        className="slds-pill__icon_container"
      >
        <span
          title="Error"
        >
          <svg
            aria-hidden="true"
            className="slds-icon-text-error slds-icon slds-icon-text-default"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#warning"
            />
          </svg>
          
        </span>
      </span>
      <a
        className="slds-pill__action"
        href="javascript:void(0);"
        onClick={[Function]}
        title="Full pill label verbiage mirrored here"
      >
        <span
          className="slds-pill__label"
        >
          Pill Label
        </span>
      </a>
      <span
        className="slds-icon_container slds-pill__remove"
        onClick={[Function]}
        role="button"
        title="Remove"
      >
        <svg
          className="slds-icon slds-icon_x-small slds-icon-text-default"
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Remove
        </span>
      </span>
    </span>
  </div>
</div>
`;

exports[`Icon, Avatar, Error HTML Snapshot 1`] = `
"<div class=\\"slds-grid slds-grid_pull-padded-medium\\">
  <div class=\\"slds-p-horizontal_medium\\"><span role=\\"button\\" class=\\"slds-pill slds-pill_link\\"><span class=\\"slds-pill__icon_container\\"><span class=\\"slds-icon_container slds-icon-standard-account\\" title=\\"Account\\"><svg aria-hidden=\\"true\\" class=\\"slds-icon\\"><use xlink:href=\\"/assets/icons/standard-sprite/svg/symbols.svg#account\\"></use></svg></span></span>
    <a
      href=\\"javascript:void(0);\\" class=\\"slds-pill__action\\" title=\\"Full pill label verbiage mirrored here\\"><span class=\\"slds-pill__label\\">Pill Label</span></a><span class=\\"slds-icon_container slds-pill__remove\\" title=\\"Remove\\" role=\\"button\\"><svg style=\\"cursor:pointer\\" class=\\"slds-icon slds-icon_x-small slds-icon-text-default\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#close\\"></use></svg><span class=\\"slds-assistive-text\\">Remove</span></span>
      </span>
  </div>
  <div class=\\"slds-p-horizontal_medium\\"><span role=\\"button\\" class=\\"slds-pill slds-pill_link\\"><span class=\\"slds-pill__icon_container\\"><div><span class=\\"slds-avatar slds-avatar_circle slds-avatar_medium\\"><img alt=\\"\\" src=\\"https://lightningdesignsystem.com/assets/images/avatar2.jpg\\" title=\\"User avatar\\"/></span></div>
  </span><a href=\\"javascript:void(0);\\" class=\\"slds-pill__action\\" title=\\"Full pill label verbiage mirrored here\\"><span class=\\"slds-pill__label\\">Pill Label</span></a><span class=\\"slds-icon_container slds-pill__remove\\" title=\\"Remove\\" role=\\"button\\"><svg style=\\"cursor:pointer\\" class=\\"slds-icon slds-icon_x-small slds-icon-text-default\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#close\\"></use></svg><span class=\\"slds-assistive-text\\">Remove</span></span>
  </span>
</div>
<div class=\\"slds-p-horizontal_medium\\"><span role=\\"button\\" class=\\"slds-pill slds-pill_link slds-has-error\\"><span class=\\"slds-pill__icon_container\\"><span title=\\"Error\\"><svg aria-hidden=\\"true\\" class=\\"slds-icon-text-error slds-icon slds-icon-text-default\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#warning\\"></use></svg></span></span>
  <a
    href=\\"javascript:void(0);\\" class=\\"slds-pill__action\\" title=\\"Full pill label verbiage mirrored here\\"><span class=\\"slds-pill__label\\">Pill Label</span></a><span class=\\"slds-icon_container slds-pill__remove\\" title=\\"Remove\\" role=\\"button\\"><svg style=\\"cursor:pointer\\" class=\\"slds-icon slds-icon_x-small slds-icon-text-default\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#close\\"></use></svg><span class=\\"slds-assistive-text\\">Remove</span></span>
    </span>
</div>
</div>"
`;

exports[`Linked, Unlinked, Truncated DOM Snapshot 1`] = `
<div
  className="slds-grid slds-grid_pull-padded-medium"
>
  <div
    className="slds-p-horizontal_medium"
  >
    <span
      className="slds-pill slds-pill_link"
      onClick={null}
      onKeyDown={[Function]}
      role="button"
    >
      <a
        className="slds-pill__action"
        href="javascript:void(0);"
        onClick={[Function]}
        title="Full pill label verbiage mirrored here"
      >
        <span
          className="slds-pill__label"
        >
          Pill Label
        </span>
      </a>
      <span
        className="slds-icon_container slds-pill__remove"
        onClick={[Function]}
        role="button"
        title="Remove"
      >
        <svg
          className="slds-icon slds-icon_x-small slds-icon-text-default"
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Remove
        </span>
      </span>
    </span>
  </div>
  <div
    className="slds-p-horizontal_medium"
  >
    <span
      className="slds-pill slds-pill_link"
      onClick={null}
      onKeyDown={[Function]}
      role="button"
    >
      <a
        className="slds-pill__action"
        href="javascript:void(0);"
        title="Full pill label verbiage mirrored here"
      >
        <span
          className="slds-pill__label"
        >
          Pill Label
        </span>
      </a>
      <span
        className="slds-icon_container slds-pill__remove"
        onClick={[Function]}
        role="button"
        title="Remove"
      >
        <svg
          className="slds-icon slds-icon_x-small slds-icon-text-default"
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Remove
        </span>
      </span>
    </span>
  </div>
  <div
    className="slds-p-horizontal_medium"
  >
    <div
      style={
        Object {
          "position": "relative",
          "width": "220px",
        }
      }
    >
      <div
        className="slds-pill_container"
      >
        <span
          className="slds-pill slds-pill_link"
          onClick={null}
          onKeyDown={[Function]}
          role="button"
        >
          <a
            className="slds-pill__action"
            href="javascript:void(0);"
            onClick={[Function]}
            title="Pill label that is longer than the area that contains it"
          >
            <span
              className="slds-pill__label"
            >
              Pill label that is longer than the area that contains it
            </span>
          </a>
          <span
            className="slds-icon_container slds-pill__remove"
            onClick={[Function]}
            role="button"
            title="Remove"
          >
            <svg
              className="slds-icon slds-icon_x-small slds-icon-text-default"
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              <use
                xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
              />
            </svg>
            <span
              className="slds-assistive-text"
            >
              Remove
            </span>
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Linked, Unlinked, Truncated HTML Snapshot 1`] = `
"<div class=\\"slds-grid slds-grid_pull-padded-medium\\">
  <div class=\\"slds-p-horizontal_medium\\"><span role=\\"button\\" class=\\"slds-pill slds-pill_link\\"><a href=\\"javascript:void(0);\\" class=\\"slds-pill__action\\" title=\\"Full pill label verbiage mirrored here\\"><span class=\\"slds-pill__label\\">Pill Label</span></a><span class=\\"slds-icon_container slds-pill__remove\\"
      title=\\"Remove\\" role=\\"button\\"><svg style=\\"cursor:pointer\\" class=\\"slds-icon slds-icon_x-small slds-icon-text-default\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#close\\"></use></svg><span class=\\"slds-assistive-text\\">Remove</span></span>
    </span>
  </div>
  <div class=\\"slds-p-horizontal_medium\\"><span role=\\"button\\" class=\\"slds-pill slds-pill_link\\"><a href=\\"javascript:void(0);\\" class=\\"slds-pill__action\\" title=\\"Full pill label verbiage mirrored here\\"><span class=\\"slds-pill__label\\">Pill Label</span></a><span class=\\"slds-icon_container slds-pill__remove\\"
      title=\\"Remove\\" role=\\"button\\"><svg style=\\"cursor:pointer\\" class=\\"slds-icon slds-icon_x-small slds-icon-text-default\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#close\\"></use></svg><span class=\\"slds-assistive-text\\">Remove</span></span>
    </span>
  </div>
  <div class=\\"slds-p-horizontal_medium\\">
    <div style=\\"width:220px;position:relative\\">
      <div class=\\"slds-pill_container\\"><span role=\\"button\\" class=\\"slds-pill slds-pill_link\\"><a href=\\"javascript:void(0);\\" class=\\"slds-pill__action\\" title=\\"Pill label that is longer than the area that contains it\\"><span class=\\"slds-pill__label\\">Pill label that is longer than the area that contains it</span></a>
        <span
          class=\\"slds-icon_container slds-pill__remove\\" title=\\"Remove\\" role=\\"button\\"><svg style=\\"cursor:pointer\\" class=\\"slds-icon slds-icon_x-small slds-icon-text-default\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#close\\"></use></svg><span class=\\"slds-assistive-text\\">Remove</span></span>
          </span>
      </div>
    </div>
  </div>
</div>"
`;
